db.py

```
@@ -15,62 +22,61 @@ def log_event(event, user_id, value, field, extra=''):
+                    on conflict do nothing
                     ''', (user_id,))
         con.commit()
         if field in ('name', 'voice_time', 'voice_counter', 'beta'):
-            con.execute(f'''
+            cur.execute(f'''
                         UPDATE users
-                        SET {field} = ?
-                        WHERE tg_id = ?
+                        SET {field} = %s
+                        WHERE tg_id = %s
                         ''', (value, user_id))
```

Ruff (0.14.3)

Possible SQL injection vector through string-based query construction

(S608)